File: frontend/src/types/ticket.ts
export interface TicketType {
  id: string;
  eventId: string;
  name: string;
  description?: string;
  price: number;
  currency: string;
  totalQuantity: number;
  soldQuantity: number;
  minPerOrder: number;
  maxPerOrder: number;
  type: 'Standard' | 'VIP' | 'EarlyBird' | 'Group';
  salesStartAt?: string;
  salesEndAt?: string;
  isActive: boolean;
  displayOrder: number;
}

export interface Ticket {
  id: string;
  orderItemId: string;
  orderId: string;
  eventId: string;
  userId: string;
  ticketNumber: string;
  qrCodeData: string;
  qrCodeImageUrl?: string;
  status: 'Active' | 'Used' | 'Cancelled' | 'Refunded';
  checkedInAt?: string;
  event?: { title: string; startDate: string; venue?: { name: string } };
  ticketTypeName?: string;
}

export interface Order {
  id: string;
  userId: string;
  discountCodeId?: string;
  orderNumber: string;
  subtotalAmount: number;
  discountAmount: number;
  taxAmount: number;
  totalAmount: number;
  currency: string;
  status: 'Pending' | 'Confirmed' | 'Cancelled' | 'Expired';
  paymentMethod?: string;
  paymentStatus?: string;
  createdAt: string;
  orderItems: OrderItem[];
  discountCode?: DiscountCode;
}

export interface OrderItem {
  id: string;
  orderId: string;
  ticketTypeId: string;
  ticketTypeName: string;
  quantity: number;
  unitPrice: number;
  subtotal: number;
  tickets: Ticket[];
}

export interface TicketReservation {
  id: string;
  reservationCode: string;
  ticketTypeId: string;
  quantity: number;
  expiresAt: string;
  isConfirmed: boolean;
}

export interface DiscountCode {
  id: string;
  eventId: string;
  code: string;
  description?: string;
  type: 'Percentage' | 'FixedAmount';
  value: number;
  maxTotalUses: number;
  currentUses: number;
  maxUsesPerUser: number;
  isActive: boolean;
}

export interface CreateOrderRequest {
  reservationId: string;
  discountCode?: string;
}

File: src/EventHub.Domain/Entities/Tickets/Ticket.cs
using EventHub.Domain.Entities.Events;
using EventHub.Domain.Entities.Users;

namespace EventHub.Domain.Entities.Tickets;

public sealed class Ticket
{
    public Guid Id { get; init; }
    public Guid OrderItemId { get; set; }
    public Guid OrderId { get; set; }
    public Guid EventId { get; set; }
    public Guid UserId { get; set; }
    public string TicketNumber { get; set; } = string.Empty;
    public string QrCodeData { get; set; } = string.Empty;
    public string? QrCodeImageUrl { get; set; }
    public string Status { get; set; } = "Active";
    public DateTime? CheckedInAt { get; set; }
    public Guid? CheckedInBy { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public OrderItem OrderItem { get; set; } = null!;
    public Order Order { get; set; } = null!;
    public Event Event { get; set; } = null!;
    public User User { get; set; } = null!;
    public User? CheckedInByUser { get; set; }
}